npm/ux/circuit.tsx

```diff
@@ -4,7 +4,48 @@
 import * as qviz from "@microsoft/quantum-viz.js/lib";
 import { useEffect, useRef } from "preact/hooks";
 
+// For perf reasons we set a limit on how many gates/qubits
+// we attempt to render. This is still a lot higher than a human would
+// reasonably want to look at, but it takes about a second to
+// render a circuit this big on a mid-grade laptop so we allow it.
+const MAX_OPERATIONS = 10000;
+const MAX_QUBITS = 1000;
+
+/* This component is shared by the Python widget and the VS Code panel */
 export function Circuit(props: { circuit: qviz.Circuit }) {
+  const errorDiv =
+    props.circuit.qubits.length === 0 ? (
+      <div>
+        <p>No circuit to display. The program has not allocated any qubits.</p>
```

Comment:
```suggestion
        <p>No circuit to display. The program does not allocate any qubits.</p>
```

Reply:
This is super nit but using past tense in conjunction with "any" short-circuited my brain for some reason.

Reply:
Ahh. I see the phrasing is awkward if you're just generating a circuit for a program. :(

I was thinking of the debugging scenario when I wrote this. If you're stepping through a program during debugging, this is the first thing you see. We don't have any qubits _yet_, but they may appear as the program continues. So present tense doesn't quite work here either.

I'll try to think of a wording that works in both scenarios...